dist/videojs-errors.js

```diff
@@ -140,14 +138,16 @@
 
     // PLAYER_ERR_NO_SRC
     player.on('play', function() {
-      if (player.currentSrc() === null ||
-          player.currentSrc() === undefined ||
-          player.currentSrc() === '') {
-        player.error({
-          code: -1,
-          type: 'PLAYER_ERR_NO_SRC'
-        });
-      }
+      window.setTimeout(function(){
+        if (player.currentSrc() === null ||
+            player.currentSrc() === undefined ||
+            player.currentSrc() === '') {
+          player.error({
+            code: -1,
+            type: 'PLAYER_ERR_NO_SRC'
+          });
+        }
+      }, 500);
```

Comment:
I don't think it's an error to create a player that's in autoplay mode but doesn't have any content at all. For instance, you could create a page filled with image thumbnails and load them into a single autoplaying player instance when the user clicks on one of them (which might be more than 500ms). Should video.js be firing a play event for autoplay when there is no source set?